samples/Shared/UnitTestFramework/GenericAnalyzerTest.cs

```diff
@@ -0,0 +1,1355 @@
+        /// Gets a collection of diagnostics to explicitly disable in the <see cref="CompilationOptions"/> for projects.
+        /// </summary>
+        public List<string> DisabledDiagnostics { get; } = new List<string>();
+
+        /// <summary>
+        /// Gets or sets the index of the code fix to apply.
+        /// </summary>
+        public int? CodeFixIndex { get; set; }
+
+        /// <summary>
+        /// Sets the expected output source file for code fix testing.
+        /// </summary>
+        /// <seealso cref="FixedSources"/>
+        public string FixedCode
+        {
+            set
+            {
+                Assert.Empty(FixedSources);
+                if (value != null)
+                {
+                    FixedSources.Add(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the set of expected output files for code fix testing. Files may be added to this list using one of the
+        /// <see cref="SourceFileList.Add"/> methods.
+        /// </summary>
+        public SourceFileList FixedSources { get; }
+
+        /// <summary>
+        /// Sets the expected output source file after a Fix All operation is applied.
+        /// </summary>
+        /// <seealso cref="BatchFixedSources"/>
+        public string BatchFixedCode
+        {
+            set
+            {
+                Assert.Empty(BatchFixedSources);
+                if (value != null)
+                {
+                    BatchFixedSources.Add(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the set of expected output files after a Fix All operation is applied. Files may be added to this list
+        /// using one of the <see cref="SourceFileList.Add"/> methods.
+        /// </summary>
+        /// <remarks>
+        /// <para>By default, Fix All operations are expected to produce the same result as incremental fix operations.
+        /// If this collection is not specified for the test, <see cref="FixedSources"/> provides the expected test
+        /// results for both incremental and Fix All scenarios.</para>
+        /// </remarks>
+        public SourceFileList BatchFixedSources { get; }
+
+        /// <summary>
+        /// Gets or sets the number of code fix iterations expected during code fix testing.
+        /// </summary>
+        /// <remarks>
+        /// <para>Code fixes are applied until one of the following conditions are met:</para>
+        ///
+        /// <list type="bullet">
+        /// <item><description>No diagnostics are reported in the input.</description></item>
+        /// <item><description>No code fixes are provided for the diagnostics reported in the input.</description></item>
+        /// <item><description>The code fix applied for the diagnostics does not produce a change in the source file(s).</description></item>
+        /// <item><description>The maximum number of allowed iterations is exceeded.</description></item>
```

Comment:
@dotnet/csharplangdesign This is less pleasant than it need to be: dotnet/csharplang#1765